compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -442,24 +445,55 @@ internal class SyntheticAccessorLowering(val context: JvmBackendContext) : IrEle
         }
     }
 
-    // Counter used to disambiguate accessors to functions with the same base name.
-    private var nameCounter = 0
-
-    private fun IrFunction.accessorName(): Name {
+    private fun IrFunction.accessorName(superQualifier: IrClassSymbol?): Name {
         val jvmName = context.methodSignatureMapper.mapFunctionName(this)
-        return Name.identifier("access\$$jvmName\$${nameCounter++}")
+        val suffix = when {
+            // The only function accessors placed on interfaces are for private functions and JvmDefault implementations.
+            // The two cannot clash.
+            parentAsClass.isJvmInterface && Visibilities.isPrivate(visibility) -> ""
```

Comment:
Not sure, but maybe just check visibility otherwise it would be default case anyway
```parentAsClass.isJvmInterface && !isPrivate(visibility) -> "\jd"``` ?

Reply:
Then it will hit the `isStatic` case, though. I am not too certain on any of these, but they are either right or the test-suite does not have sufficient coverage ;)

Reply:
Yes, indeed )